### ISSUE TITLE
 F.poisson_nll_loss ignores  reduction='none'

### ISSUE THREAD

**Issue by untom** [pytorch/pytorch#17347]
Hi!
Using PyTorch 1.0.1, it seems like  F.poisson_nll_loss ignores the `reduction='none'`:

```
import torch
import torch.nn.functional as F
input = torch.tensor([0.5, 1.5, 2.5])
target = torch.tensor([1., 2., 3.])
print(F.poisson_nll_loss(input, target, reduction='none'))
```
gives
```tensor(7.3129)```

i.e.,  it ignores the `reduction='none'` argument.

**Comment by vishwakftw** [comment]
I've found out the issue, will send in a fix.

**Comment by SsnL** [comment]
fixed in #17358
